### PR TITLE
Workers

### PR DESCRIPTION
**Pull request by jdebr**

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
+  "compilerOptions": {
+    "allowSyntheticDefaultImports": true,
+    "noFallthroughCasesInSwitch": true,
+    "noUnusedParameters": true,
+    "noImplicitReturns": true,
+    "moduleResolution": "node",
+    "esModuleInterop": true,
+    "noUnusedLocals": true,
+    "noImplicitAny": true,
```
Comment (natenorberg) [inline tsconfig.json:10]: Way to be A+ in converting to TS

File: src/QuiqSockets.ts [hunk 2]
```diff
@@ -126,28 +130,28 @@ class QuiqSocket {
...
+    globalThis.addEventListener('online', () => {
       this._log.info('QuiqSocket online event');
       if (this._waitingForOnlineToReconnect) {
         this.connect();
       }
       this._waitingForOnlineToReconnect = false;
     });
 
-    window.addEventListener('offline', () => {
+    globalThis.addEventListener('offline', () => {
```
Comment (natenorberg) [inline src/QuiqSockets.ts:154]: Where is this defined?

File: tsconfig.json (hunk 1 above)
Comment (jdebr) [reply tsconfig.json:10]: thanks!

File: src/QuiqSockets.ts (hunk 2 above)
Comment (jdebr) [reply src/QuiqSockets.ts:154]: It's a mostly-supported native reference - no IE11 support but babel does support it, I'll need to confirm webchat agrees with this
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/globalThis

**Pull request merged**